sanic/request.py

```diff
@@ -105,6 +105,18 @@ def args(self):
 
         return self.parsed_args
 
+    @property
```

Comment:
Being lazy is helpful though in case the connection doesn't live long enough i.e. if you get the headers but when you start receiving the body it is too long. Then there's no point calculating the version, URL, etc.